tileview/src/main/java/com/qozix/tileview/widgets/ZoomPanLayout.java

```diff
@@ -497,10 +520,10 @@ public void scrollTo( int x, int y ) {
   }
 
   private void calculateMinimumScaleToFit() {
-    if( mShouldScaleToFit ) {
+    if (mMinimumScaleMode != MinimumScaleMode.NONE) {
       float minimumScaleX = getWidth() / (float) mBaseWidth;
       float minimumScaleY = getHeight() / (float) mBaseHeight;
-      float recalculatedMinScale = Math.max( minimumScaleX, minimumScaleY );
+      float recalculatedMinScale = mMinimumScaleMode == MinimumScaleMode.FILL ? Math.max( minimumScaleX, minimumScaleY ) : Math.min( minimumScaleX, minimumScaleY );
```

Comment:
So if we are in `FIT` mode, `mMinimumScaleMode` is set to `Math.min( minimumScaleX, minimumScaleY )`
Maybe we should be more explicit here, because we have to understand that we are not in mode `NONE`, nor in `FILL`, so we must be in `FIT` mode.